### PR TITLE
fs/smartfs : Add journal format to signature sector, logical 0

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
If level of crc is same but journal state is different, guide it to lower level format

### CODE REVIEW

File: os/fs/driver/mtd/smart.c [hunk 1]
```diff
@@ -124,8 +133,9 @@
 #define SMART_FMT_SIG4            'T'
 
 #define SMART_FMT_VERSION_POS     (SMART_FMT_POS1 + 4)
-#define SMART_FMT_NAMESIZE_POS    (SMART_FMT_POS1 + 5)
-#define SMART_FMT_ROOTDIRS_POS    (SMART_FMT_POS1 + 6)
+#define SMART_FMT_JOURNAL_POS     (SMART_FMT_POS1 + 5)
```
Comment (amogh-samsung) [inline os/fs/driver/mtd/smart.c:136]: Any particular reason for adding this before NAMESIZE and ROOTDIR signature bytes?
Order of significance?

Comment (Taejun-Kwon) [reply os/fs/driver/mtd/smart.c:136]: Journaling is tightly coupled with crc, so I added here.

Comment (amogh-samsung) [reply os/fs/driver/mtd/smart.c:136]: Fair enough

**Review by sangwon03 — APPROVED** [review]
LGTM

**Pull request merged**